### PR TITLE
Feature/3527 toolbar menu examples

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3527

#### Changes proposed in this Pull Request:
- Added Toolbar Menu example

#### Screenshots:
Screenshot 2022-09-28 at 3 44 52 PM

#### To Test:
- yarn start:reactdev

### CODE REVIEW

**Review by ektasawant — COMMENTED** [review]
Do we need sx prop in code snippet block?

File: docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Box } from '@mui/material';
+import { CodeBlock, CodeBlockActionButtonRow } from '../../../shared';
+import * as colors from '@brightlayer-ui/colors';
+import { ToolbarMenuWithIconExample } from './ToolbarMenuWithIconExample';
+
+const codeSnippet = `<ToolbarMenu
+    sx={{ maxWidth: 'inherit' }}
```
Comment (ektasawant) [inline docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIcon.tsx:8]: Do we need sx prop in code snippet block?

File: docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIconExample.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import Box from '@mui/material/Box';
+import { ToolbarMenu } from '@brightlayer-ui/react-components';
+import { Home } from '@mui/icons-material';
+
+export const ToolbarMenuWithIconExample = (): JSX.Element => (
+    <Box>
+        <ToolbarMenu
+            sx={{ maxWidth: 'inherit' }}
```
Comment (ektasawant) [inline docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIconExample.tsx:9]: Do we need sx prop in code snippet block?

**Comment by huayunh** [comment]
Totally agree with Ekta.

**Comment by huayunh** [comment]
- [ ] "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
- [ ] Same example, the line highlighting started at the wrong line. It should start from the line below.
- [ ] Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
- [ ] Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon

File: docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIconExample.tsx (hunk 2 above)
Comment (bkarambe) [reply docs/src/componentDocs/ToolbarMenu/examples/ToolbarMenuWithIconExample.tsx:9]: Yes we need it in the example, but we can remove it from snippet

**Comment by bkarambe** [comment]
> * collapsed

Updated

**Comment by bkarambe** [comment]
> * [ ]  "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> * [ ]  Same example, the line highlighting started at the wrong line. It should start from the line below.
> * [ ]  Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> * [ ]  Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon

Updated

**Comment by bkarambe** [comment]
> > * [ ]   "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> > * [x]   Same example, the line highlighting started at the wrong line. It should start from the line below.
> > * [x]   Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> > * [x]   Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon
> 
> Updated

I checked the syntax highlighting of InfoListItem, there are no missing brackets

**Comment by ektasawant** [comment]
> > > * [ ]    "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> > > * [x]    Same example, the line highlighting started at the wrong line. It should start from the line below.
> > > * [x]    Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> > > * [x]    Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon
> > 
> > 
> > Updated
> 
> I checked the syntax highlighting of InfoListItem, there are no missing brackets

This is to line no 17-25 of [ToolbarMenuWithinToolbar.tsx](https://github.com/brightlayer-ui/react-component-library/pull/555/files#diff-a911bcb6e37febe04a2d06b3712c093b5f13607c3c543d4cbb9798363e66f431) 
menuGroups={[
                        {
                            items: [
                                { title: 'Menu Item 1' },
                                { title: 'Menu Item 2' },
                                { title: 'Menu Item 3' },
                            ],
                        },
                    ]}
                    
 Seems prism.js breaks the css when we have [] brackets.

**Comment by bkarambe** [comment]
> > > > * [ ]     "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> > > > * [x]     Same example, the line highlighting started at the wrong line. It should start from the line below.
> > > > * [x]     Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> > > > * [x]     Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon
> > > 
> > > 
> > > Updated
> > 
> > 
> > I checked the syntax highlighting of InfoListItem, there are no missing brackets
> 
> This is to line no 17-25 of [ToolbarMenuWithinToolbar.tsx](https://github.com/brightlayer-ui/react-component-library/pull/555/files#diff-a911bcb6e37febe04a2d06b3712c093b5f13607c3c543d4cbb9798363e66f431) menuGroups={[ { items: [ { title: 'Menu Item 1' }, { title: 'Menu Item 2' }, { title: 'Menu Item 3' }, ], }, ]}
> 
> Seems prism.js breaks the css when we have [] brackets when the language is JSX.

Yes

**Comment by ektasawant** [comment]
> > > > > * [ ]      "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> > > > > * [x]      Same example, the line highlighting started at the wrong line. It should start from the line below.
> > > > > * [x]      Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> > > > > * [x]      Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon
> > > > 
> > > > 
> > > > Updated
> > > 
> > > 
> > > I checked the syntax highlighting of InfoListItem, there are no missing brackets
> > 
> > 
> > This is to line no 17-25 of [ToolbarMenuWithinToolbar.tsx](https://github.com/brightlayer-ui/react-component-library/pull/555/files#diff-a911bcb6e37febe04a2d06b3712c093b5f13607c3c543d4cbb9798363e66f431) menuGroups={[ { items: [ { title: 'Menu Item 1' }, { title: 'Menu Item 2' }, { title: 'Menu Item 3' }, ], }, ]}
> > Seems prism.js breaks the css when we have [] brackets when the language is JSX.
> 
> Yes

@bkarambe , Can you log an issue for this?

**Comment by bkarambe** [comment]
> > > > > > * [ ]       "Toolbar Menu within Toolbar" — It looks like your code snippet syntax color highlighting is going strange, probably due to a missing parenthesis or bracket.
> > > > > > * [x]       Same example, the line highlighting started at the wrong line. It should start from the line below.
> > > > > > * [x]       Same example, please set the app bar to be always `collapsed`. If you scroll to the top of the page and quickly scroll down back to the page bottom, you will notice the app bar secretly collapsing on its own.
> > > > > > * [x]       Toolbar Menu with Icon — "optionally accepts a `icon`" -> should be "an" icon
> > > > > 
> > > > > 
> > > > > Updated
> > > > 
> > > > 
> > > > I checked the syntax highlighting of InfoListItem, there are no missing brackets
> > > 
> > > 
> > > This is to line no 17-25 of [ToolbarMenuWithinToolbar.tsx](https://github.com/brightlayer-ui/react-component-library/pull/555/files#diff-a911bcb6e37febe04a2d06b3712c093b5f13607c3c543d4cbb9798363e66f431) menuGroups={[ { items: [ { title: 'Menu Item 1' }, { title: 'Menu Item 2' }, { title: 'Menu Item 3' }, ], }, ]}
> > > Seems prism.js breaks the css when we have [] brackets when the language is JSX.
> > 
> > 
> > Yes
> 
> @bkarambe , Can you log an issue for this?

Updated this point the issue : https://github.com/brightlayer-ui/react-component-library/issues/485

**Review by ektasawant — APPROVED** [review]

**Pull request merged**